widget/numerical_entry.go

```diff
@@ -15,37 +24,131 @@ type NumericalEntry struct {
 	AllowFloat bool
 	// AllowNegative determines if negative numbers can be entered.
 	AllowNegative bool
+	minus         rune
+	radixSep      rune
+	thouSep       rune
+
+	binder basicBinder
+	locale string
+	mPr    *message.Printer
 }
 
 // NewNumericalEntry returns an extended entry that only allows numerical input.
 func NewNumericalEntry() *NumericalEntry {
 	entry := &NumericalEntry{}
+	var err error
+	entry.locale, err = jibber_jabber.DetectIETF()
+	if err != nil {
+		fyne.LogError("DetectIETF error: %s\n", err)
+		entry.locale = "en-US" // Fallback to en-US
+	}
+	lang, err := language.Parse(entry.locale)
+	if err != nil {
+		fyne.LogError("Language parse error: ", err)
+		lang = language.English // Fallback to English
+		entry.locale = "en-US"
+	}
+	entry.mPr = message.NewPrinter(lang)
+
+	entry.getLocaleRunes(entry.locale)
 	entry.ExtendBaseWidget(entry)
+	entry.Validator = entry.ValidateText
 	return entry
 }
 
+// NewNumericalEntryWithData creates a numerical entry that is bound to a
+// data source and can allow or disallow float and negative numbers.
+func NewNumericalEntryWithData(allowFloat bool, allowNegative bool, data binding.Float) *NumericalEntry {
+	e := NewNumericalEntry()
+	e.AllowFloat = allowFloat
+	e.AllowNegative = allowNegative
+	e.Bind(data)
+	e.OnChanged = func(string) {
+		e.binder.CallWithData(e.writeData)
+	}
+
+	return e
+}
+
+// Append appends text to the entry, filtering out non-numerical characters
+// based on the current locale and allowed input types (negative, float).
+func (e *NumericalEntry) Append(text string) {
+	s := e.getValidText(e.Text, text)
+	e.Entry.Append(s)
+}
+
+// Bind connects the specified data source to this Spinner widget.
+// The current value will be displayed and any changes in the data will cause the widget
+// to update.
+func (e *NumericalEntry) Bind(data binding.Float) {
+	e.binder.SetCallback(e.updateFromData)
+	e.binder.Bind(data)
+}
+
+// ParseFloat parses the text content of the entry as a float64.
+// It returns the parsed float and an error if parsing fails.
+func (e *NumericalEntry) ParseFloat() (float64, error) {
```

Comment:
I would expect from the name "ParseFloat" that it takes a parameter to parse. Maybe "Float()" sets the expectation better? it is essentially just a getter.

Reply:
Or just "Value()" to reflect the "SetValue"?

Reply:
Changed to Value().